lib/data/multipleBackendGateway.js

```diff
@@ -114,7 +118,18 @@ const multipleBackendGateway = {
                     awsArray.length)];
                 const checkThisOne = clients[randomAWS];
                 return checkThisOne.checkAWSHealth(randomAWS, multBackendResp,
-                    callback);
+                (errNull, multBackendResp) => {
+                    // do not neet to check for err !== null since it is
+                    // always null
+                    if (azureArray.length > 0) {
```

Comment:
if you only have azure backends but not aws backends, you will never do these checks. you could do the aws and azure checks in parallel instead.